src/3d/qgsoffscreen3dengine.cpp

```diff
@@ -238,17 +143,11 @@ QSurface *QgsOffscreen3DEngine::surface() const
 
 void QgsOffscreen3DEngine::requestCaptureImage()
 {
-  if ( mReply )
-  {
-    QgsDebugMsgLevel( QStringLiteral( "already having a pending capture, skipping" ), 2 );
-    return;
-  }
-  mReply = mRenderCapture->requestCapture();
-  connect( mReply, &Qt3DRender::QRenderCaptureReply::completed, this, [ = ]
+  Qt3DRender::QRenderCaptureReply *captureReply;
+  captureReply = mFrameGraph->renderCapture()->requestCapture();
+  connect( captureReply, &Qt3DRender::QRenderCaptureReply::completed, this, [ = ]
   {
-    QImage image = mReply->image();
-    mReply->deleteLater();
-    mReply = nullptr;
-    emit imageCaptured( image );
+    emit imageCaptured( captureReply->image() );
+    captureReply->deleteLater();
```

Comment:
Just wondering, does this change fix anything regarding capture of images, or it is only changed to remove `mReply` from member variables?

Reply:
It is changed to make things the same between QgsWindow3DEngine and QgsOffscreen3DEngine, should I move it up to QgsAbstract3DEngine ?

Reply:
Hmm I see... If both engines support capturing of renders, then maybe it's best to have the implementation in the parent class (especially if the implementations are the same)

Reply:
The implementation is the same and it is moved up now.